### PR TITLE
add unit tests for observer

### PR DESCRIPTION
**Pull request by joelberkeley**

### CODE REVIEW

File: tests/unit/test_observer.py [hunk 1]
```diff
@@ -11,30 +11,43 @@
...
 
+from trieste.data import Dataset
 from trieste.observer import map_is_finite, filter_finite
 
+from tests.util.misc import ShapeLike, assert_datasets_allclose
+
 
-def _nan_at_origin(t: tf.Tensor) -> tf.Tensor:
+def _sum_with_nan_at_origin(t: tf.Tensor) -> tf.Tensor:
     is_at_origin = tf.reduce_all(t == [[0., 0.]], axis=-1, keepdims=True)
```
Comment (ChrisMorter) [inline tests/unit/test_observer.py:37]: A short comment saying what this method does may be helpful here. The function name alone didn't quite do it for me.

File: tests/unit/test_observer.py [hunk 2]
```diff
@@ -11,30 +11,43 @@
...
+        Dataset(
+            tf.constant([[-1., 0.], [1., 0.], [0., 2.], [1., 3.]]),
+            tf.constant([[-1.], [1.], [2.], [4.]])
+        )
+    ),
+    (tf.constant([[0., 0.]]), Dataset(tf.zeros([0, 2]), tf.zeros([0, 1]))),  # only failure points
+    (tf.zeros([0, 2]), Dataset(tf.zeros([0, 2]), tf.zeros([0, 1]))),  # empty data
+])
+def test_filter_finite(query_points: tf.Tensor, expected: Dataset) -> None:
+    actual = filter_finite(query_points, _sum_with_nan_at_origin(query_points))
```
Comment (ChrisMorter) [inline tests/unit/test_observer.py]: I feel extracting a variable for `_sum_with_nan_at_origin(query_points)`, named `observations` would make this easier to read.

File: tests/unit/test_observer.py [hunk 3]
```diff
@@ -43,14 +56,20 @@ def test_filter_finite() -> None:
...
+def test_filter_finite_raises_for_invalid_shapes(qp_shape: ShapeLike, obs_shape: ShapeLike) -> None:
     with pytest.raises(ValueError):
         filter_finite(tf.ones(qp_shape), tf.ones(obs_shape))
 
 
 def test_map_is_finite() -> None:
-    query_points = tf.constant([[0., 0.]] + [[-1., 0.], [1., 0.], [0., 2.], [1., 3.]])
-    is_finite = map_is_finite(query_points, _nan_at_origin(query_points))
+    query_points = tf.constant([[-1., 0.], [1., 0.], [0., 2.], [0., 0.], [1., 3.]])
+    is_finite = map_is_finite(query_points, _sum_with_nan_at_origin(query_points))
```
Comment (ChrisMorter) [inline tests/unit/test_observer.py]: Same comment as above regarding extracting a named variable for the observations (same for the test below too).

**Review by ChrisMorter — APPROVED** [review]
Looks good, just a couple of minor comments that may help improve test readability.

File: tests/unit/test_observer.py (hunk 1 above)
Comment (joelberkeley) [reply tests/unit/test_observer.py:37]: will do

File: tests/unit/test_observer.py (hunk 2 above)
Comment (joelberkeley) [reply tests/unit/test_observer.py]: will do

File: tests/unit/test_observer.py (hunk 3 above)
Comment (joelberkeley) [reply tests/unit/test_observer.py]: will do

**Pull request merged**